### ISSUE TITLE
Second ncspot instance overwrites original ipc socket

### ISSUE THREAD

**Issue by ThomasFrans** [hrkfdn/ncspot#1158]
**Describe the bug**
When opening a second instance of ncspot while one is already running, it will overwrite the IPC socket of the original instance.

**To Reproduce**
1. Open one ncspot instance
2. Connect with netcat
3. Open another instance
4. Close that second instance again
5. Connect with netcat to the socket
The socket no longer works as it is overwritten by the second instance

**Expected behavior**
I don't know what the expected behavior would be. I just thought that the current behavior can't be what would be expected. One solution I could imagine is doing something similar compared to programs like Firefox or VLC, which will create multiple MPRIS objects when multiple instances are playing media. The 'hard' or 'unclear' problem with this is that clients wouldn't really know which is the main instance and which is the second one.

**System (please complete the following information):**
 - OS: Arch Linux (latest updates)
 - Terminal: Alacritty
 - Version: main branch
 - Installed from: cargo run

**Additional context**
This issue probably only applies to developers as there wouldn't be a reason to have two instances of ncspot open at the same time. I just created this issue to keep track of this bug/unexpected behavior. It's nothing major. It would just be nice for ncspot contributors relying on the IPC implementation (not me) to have it keep working when working on ncspot as well :slightly_smiling_face: Some users could also stumble upon this when they accidentally open up a second instance and close it again.

**Comment by hrkfdn** [comment]
Hmm, I guess we could either ..
- prevent multiple instances
- use unique socket names, i.e. by including process ID
- only have the initial instance setup IPC

Any thoughts on this?

**Comment by ThomasFrans** [comment]
I also think having multiple instances is handy. How else am I supposed to listen to music when hacking on ncspot 😄 I think the most correct solution would be to create IPC sockets for every instance using something like the process id , since a running instance shouldn't influence the correct working of another instance's IPC socket. The other option would of course also be good.

**Comment by Bettehem** [comment]
Yeah I agree, allowing multiple instances and just adding the PID would probably be the best option

**Comment by hrkfdn** [comment]
It seems like a bit of an edge case as @ThomasFrans pointed, since the socket name of the subsequent ncspot instances changes and thus maybe isn't useful to the user, but it's supported now :)
